javaparser-core/src/main/java/com/github/javaparser/TokenTypes.java

```diff
@@ -22,12 +30,18 @@ public static boolean isEndOfLineCharacter(int tokenType) {
         }
     }
 
-    public static boolean isWhitespaceOrComment(int tokenType) {
-        return isWhitespace(tokenType) || isComment(tokenType);
+    public static boolean isWhitespaceOrComment(int kind) {
+        switch (getCategory(kind)) {
```

Comment:
nice. Maybe we can have a `isWhitespaceOrComment` method on the category and change this to:

`return getCategory(kind).isWhitespaceOrComment()`